pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -621,14 +594,8 @@ func (t *target) needsRemediation(timeoutForMachineToHaveNode time.Duration) (bo
 	return false, minDuration(nextCheckTimes), nil
 }
 
-func (t *target) hasMachineSetOwner() bool {
-	ownerRefs := t.Machine.ObjectMeta.GetOwnerReferences()
-	for _, or := range ownerRefs {
-		if or.Kind == ownerControllerKind {
-			return true
-		}
-	}
-	return false
+func (t *target) hasOwner() bool {
+	return len(t.Machine.ObjectMeta.GetOwnerReferences()) > 0
```

Comment:
Based on the BZ, I think this should be checking that at least one of the owner references sets the `controller` field to `true`. Then we know there's some controller looking after this resource somewhere

Should double check that MachineSets set controller references as well, they should be if they aren't